routers/api/v2/tokens.go

```diff
@@ -53,14 +46,51 @@ func (r *apiV2Router) CreateServiceToken(ctx *gin.Context) {
 		}
 	}
 
-	token, err := r.authorizer.CreateServiceToken(req.Owner, parsedURIs, req.ExpiresAt)
+	userToken := r.GetAuthToken(ctx)
+	userID, err := r.authorizer.GetUserIdFromToken(userToken)
+	if err != nil {
+		r.logger.Debug("could not extract id from auth token", zap.Error(err))
+		models.SendAPIError(ctx, http.StatusBadRequest, "could not extract user id from auth token")
+		return
+	}
+
+	tokenID := primitive.NewObjectID()
+	token, err := r.authorizer.CreateServiceToken(tokenID, parsedURIs, req.ExpiresAt)
 	if err != nil {
 		r.logger.Error("could not create JWT", zap.Error(err))
 		models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
 		return
 	}
 
+	_, err = r.tokenService.AddServiceToken(ctx, tokenID, userID.Hex(), token)
```

Comment:
Storing service tokens is specific to the authorization system, not the v2 API controller. Token persistence should be handled within `Authorizer.CreateServiceToken`

Reply:
Good point, I'll move the logic 🙂